Выборы в США - 2

Maksonchik88 opened the pull request:
https://informatics.msk.ru/mod/statements/view.php?id=4535&chapterid=3770#1

*Инициализировал словарь штат -> кол-во выборщиков 
*Инициализировал словарь штаты -> {кандидаты: количество } 
*Отсортировал значения второго словаря, согласно условию задачи

 *Не убрал отладочные принты
*Не разобрался, как вывести на печать кандидата без голосовo

Maksonchik88 committed:
Выборы в США - 2

https://informatics.msk.ru/mod/statements/view.php?id=4535&chapterid=3770#1

*Инициализировал словарь штат -> кол-во выборщиков
*Инициализировал словарь штаты -> {кандидаты: количество }
*Отсортировал значения второго словаря, согласно условию задачи
*Не убрал отладочные принты
*Не разобрался, как вывести на печать кандидата без голосовo

Maksonchik88 committed:
пустой

iliadmitriev reviewed (changes requested):
Задача решена не верно.
По идее ты посчитал в каждом штате количество голосов у кандидата. (это все лишь промежуточный вариант для того чтобы определить кто победил в данном штате)
А вот выводить тебе надо список кандидатов по общему количеству голосов выборщиков.
Которое еще надо посчитать.

Maksonchik88 commented:
Почему?
Там суть в том, что у кого больше голосов в штате , тому по итогу и отдаются все голоса в количестве
Или я не так понял?
Да, я посчитал голоса каждого кандидата по штату, затем отсортировал по убыванию голосов (и по лексикографии кандидатов). 
Так вот: все голоса на штат отдаются тому, у кого голосов из данных в словаре больше. 
Я так и сделал.
Или я неправильно понял всё?

iliadmitriev commented:
> Почему? Там суть в том, что у кого больше голосов в штате , тому по итогу и отдаются все голоса в количестве Или я не так понял? Да, я посчитал голоса каждого кандидата по штату, затем отсортировал по убыванию голосов (и по лексикографии кандидатов). Так вот: все голоса на штат отдаются тому, у кого голосов из данных в словаре больше. Я так и сделал. Или я неправильно понял всё?

Да все верно сделал что посчитал по штатам, но это же просто промежуточный результат, который нужен для того чтобы посчитать финальный результат. Ты просто не доделал задачу.

Maksonchik88 committed:
Родословная: подсчет уровней

https://informatics.msk.ru/mod/statements/view.php?id=4535&chapterid=3771#1

*Прочитал файл, закрыл файл
*Инициализировал словарь и заполнил его предок -> потомок (связи)
*Создал так же список вершин дерева
*Инициализировал переменную "очередь" с родоначальником
*Инициализировал новый словарь, куда помещал правитель -> глубина
*Распечатал отсортированный новый словарь, согласно условию.

P.S.
Не создал новую ветку, сука

Maksonchik88 committed:
пусто
